lib/phonelib/phone_analyzer.rb

```diff
@@ -106,7 +106,7 @@ def detect_and_parse(phone)
       result = {}
       Phonelib.phone_data.each do |key, data|
         parsed = parse_single_country(phone, data)
-        if double_prefix_allowed?(data, phone, parsed && parsed[key])
+        if double_prefix_allowed?(data, phone, parsed && parsed[key]) && key == Phonelib.default_country.to_s.upcase
```

Comment:
@murtas Actually you are suggesting to do a double country prefix check only in case that country was specified, right? I guess such behavior should be allowed by config variable.

Reply:
> Actually you are suggesting to do a double country prefix check only in case that country was specified, right?

Correct. `strict_double_prefix_check` was added to control the new behavior.